src/api/helpers/map-and-filter-address/index.ts

```diff
@@ -0,0 +1,31 @@
+import { OrdnanceSurveyResponse, Address } from '../../types';
+import mapAddress from '../map-address';
+
+/**
+ * mapAndFilterAddress
+ * filters address based on house number or house name
+ * maps filtered address to correct format
+ * returns array of matching addresses
+ * @param {String} house name or address
+ * @param {Array<OrdnanceSurveyResponse>} ordnanceSurveyResponse
+ * @returns {Array<Address>} mapped addresses
+ */
+const mapAndFilterAddress = (house: string, ordnanceSurveyResponse: Array<OrdnanceSurveyResponse>) => {
+  // returns array of addresses which match house number or housename
+  const filtered = ordnanceSurveyResponse.filter((eachAddress) => eachAddress.DPA.BUILDING_NUMBER === house || eachAddress.DPA.BUILDING_NAME === house);
```

Comment:
Can the `house` variable be more descriptive? Should it be `houseNameOrNumber`?

Reply:
changed to houseNameOrNumber

Reply:
Param is still `house` 😛